File: update_landing.py
import re

html_path = 'f:/PROMPTLAB/public/landing.html'
with open(html_path, 'r', encoding='utf-8') as f:
    html = f.read()

new_section = """        <!-- ── Model Showcase ───────────────────────────────────── -->
        <section
            class="relative z-10 py-24 px-4 bg-slate-50 dark:bg-[#020202] border-y border-slate-200 dark:border-white/5 transition-colors duration-500 overflow-hidden"
            id="models">
            <div class="absolute inset-0 grid-overlay pointer-events-none opacity-10 dark:opacity-40 transition-opacity duration-500"></div>
            <div class="absolute top-[-10%] left-1/2 -translate-x-1/2 w-[1000px] h-[600px] bg-primary/5 dark:bg-primary/10 blur-[120px] rounded-full pointer-events-none transition-colors duration-500"></div>
            <div class="absolute bottom-[-10%] left-1/4 w-[600px] h-[400px] bg-primary/5 dark:bg-primary/5 blur-[100px] rounded-full pointer-events-none transition-colors duration-500"></div>
            
            <div class="relative z-10 max-w-7xl mx-auto flex flex-col items-center text-center">
                <div class="inline-flex items-center gap-2 px-4 py-1.5 rounded-full border border-primary/20 dark:border-primary/30 bg-primary/5 mb-10 transition-colors duration-500">
                    <span class="w-1.5 h-1.5 rounded-full bg-primary animate-pulse"></span>
                    <span class="text-[10px] font-bold tracking-[0.2em] uppercase text-primary">Model Support: Active Analysis</span>
                </div>
                <h2 class="text-4xl md:text-6xl lg:text-8xl font-black tracking-tight text-slate-900 dark:text-white mb-8 leading-none transition-colors duration-500">
                    OPTIMIZED FOR<br/>
                    <span class="text-primary glow-text-red">EVERY MAJOR AI MODEL</span>
                </h2>
                <p class="max-w-2xl mx-auto text-lg text-slate-500 dark:text-white/40 mb-20 leading-relaxed font-light transition-colors duration-500">
                    Each model has unique scoring weights, anti-patterns, and rewrite strategies tailored to how it actually works.
                </p>

                <div class="grid grid-cols-1 md:grid-cols-3 gap-8 mb-24 w-full">
                    
                    <!-- OpenAI GPT -->
                    <div class="group flex flex-col p-10 rounded-[24px] text-left transition-all duration-500 bg-white dark:bg-[#0f0f0f]/40 backdrop-blur-xl border border-slate-200 dark:border-white/5 shadow-xl dark:shadow-none hover:border-[#10A37F]/30 hover:shadow-[0_0_30px_rgba(16,185,129,0.05)] dark:hover:shadow-[0_0_30px_rgba(16,185,129,0.05)]">
                        <div class="w-14 h-14 rounded-xl bg-[#10A37F]/10 flex items-center justify-center mb-10 border border-[#10A37F]/20 group-hover:bg-[#10A37F]/20 transition-all overflow-hidden p-1.5">
                            <img src="/assets/logos/chatgpt.png" alt="ChatGPT Logo" class="w-full h-full object-contain dark:invert" />
                        </div>
                        <h3 class="text-2xl font-bold text-slate-900 dark:text-white mb-2 transition-colors duration-500">OpenAI GPT</h3>
                        <p class="text-[9px] font-bold tracking-[0.2em] uppercase text-[#10A37F] mb-10">Instruction-Following Powerhouse</p>
                        
                        <ul class="space-y-4 mb-14 flex-1">
                            <li class="flex items-center gap-3 text-sm text-slate-600 dark:text-white/70 transition-colors duration-500">
                                <span class="w-1.5 h-1.5 rounded-full bg-[#10A37F]"></span> System-role framing emphasis
                            </li>
                            <li class="flex items-center gap-3 text-sm text-slate-600 dark:text-white/70 transition-colors duration-500">
                                <span class="w-1.5 h-1.5 rounded-full bg-[#10A37F]"></span> Chain-of-thought triggers
                            </li>
                            <li class="flex items-center gap-3 text-sm text-slate-600 dark:text-white/70 transition-colors duration-500">
                                <span class="w-1.5 h-1.5 rounded-full bg-[#10A37F]"></span> Output format adherence
                            </li>
                            <li class="flex items-center gap-3 text-sm text-slate-800 dark:text-white font-medium transition-colors duration-500">
                                <span class="w-1.5 h-1.5 rounded-full bg-[#10A37F]"></span> 30% weight on controllability
                            </li>
                        </ul>
                        
                        <div class="h-1 w-full bg-slate-100 dark:bg-white/5 rounded-full flex overflow-hidden transition-colors duration-500">
                            <div class="h-full bg-primary w-[15%]"></div>
                            <div class="h-full bg-amber-500 w-[20%]"></div>
                            <div class="h-full bg-[#10A37F] w-[45%]"></div>
                            <div class="h-full bg-slate-200 dark:bg-white/10 w-[20%]"></div>
                        </div>
                    </div>

                    <!-- Anthropic Claude -->
                    <div class="group flex flex-col p-10 rounded-[24px] text-left transition-all duration-500 bg-primary/[0.02] dark:bg-primary/[0.04] backdrop-blur-xl border border-primary/20 dark:border-primary/20 shadow-xl dark:shadow-none hover:border-[#D97706]/30 hover:shadow-[0_0_30px_rgba(217,119,6,0.05)] dark:hover:shadow-[0_0_30px_rgba(217,119,6,0.05)] relative md:-translate-y-4">
                        <div class="absolute -top-3 right-8 bg-primary px-3 py-1 rounded text-[8px] font-black uppercase tracking-widest text-white shadow-lg shadow-primary/20">
                            Most Nuanced
                        </div>
                        <div class="w-14 h-14 rounded-xl bg-[#D97706]/10 flex items-center justify-center mb-10 border border-[#D97706]/20 group-hover:bg-[#D97706]/20 transition-all overflow-hidden p-1.5">
                            <img src="/assets/logos/claude.png" alt="Claude Logo" class="w-full h-full object-contain" />
                        </div>
                        <h3 class="text-2xl font-bold text-slate-900 dark:text-white mb-2 transition-colors duration-500">Anthropic Claude</h3>
                        <p class="text-[9px] font-bold tracking-[0.2em] uppercase text-[#D97706] mb-10">Safety-Aligned Precision Engine</p>
                        
                        <ul class="space-y-4 mb-14 flex-1">
                            <li class="flex items-center gap-3 text-sm text-slate-600 dark:text-white/70 transition-colors duration-500">
                                <span class="w-1.5 h-1.5 rounded-full bg-[#D97706]"></span> XML tag structure emphasis
                            </li>
                            <li class="flex items-center gap-3 text-sm text-slate-600 dark:text-white/70 transition-colors duration-500">
                                <span class="w-1.5 h-1.5 rounded-full bg-[#D97706]"></span> Explicit scope boundaries
                            </li>
                            <li class="flex items-center gap-3 text-sm text-slate-600 dark:text-white/70 transition-colors duration-500">
                                <span class="w-1.5 h-1.5 rounded-full bg-[#D97706]"></span> Safety-aligned framing
                            </li>
                            <li class="flex items-center gap-3 text-sm text-slate-800 dark:text-white font-medium transition-colors duration-500">
                                <span class="w-1.5 h-1.5 rounded-full bg-[#D97706]"></span> 30% weight on constraints
                            </li>
                        </ul>
                        
                        <div class="h-1 w-full bg-slate-100 dark:bg-white/5 rounded-full flex overflow-hidden transition-colors duration-500">
                            <div class="h-full bg-primary w-[10%]"></div>
                            <div class="h-full bg-[#D97706] w-[55%]"></div>
                            <div class="h-full bg-slate-200 dark:bg-white/10 w-[35%]"></div>
                        </div>
                    </div>

                    <!-- Google Gemini -->
                    <div class="group flex flex-col p-10 rounded-[24px] text-left transition-all duration-500 bg-white dark:bg-[#0f0f0f]/40 backdrop-blur-xl border border-slate-200 dark:border-white/5 shadow-xl dark:shadow-none hover:border-[#3B82F6]/30 hover:shadow-[0_0_30px_rgba(59,130,246,0.05)] dark:hover:shadow-[0_0_30px_rgba(59,130,246,0.05)]">
                        <div class="w-14 h-14 rounded-xl bg-[#3B82F6]/10 flex items-center justify-center mb-10 border border-[#3B82F6]/20 group-hover:bg-[#3B82F6]/20 transition-all overflow-hidden p-[2px]">
                            <img src="/assets/logos/gemini.png" alt="Gemini Logo" class="w-full h-full object-contain" />
                        </div>
                        <h3 class="text-2xl font-bold text-slate-900 dark:text-white mb-2 transition-colors duration-500">Google Gemini</h3>
                        <p class="text-[9px] font-bold tracking-[0.2em] uppercase text-[#3B82F6] mb-10">Multimodal &amp; Context-First</p>
                        
                        <ul class="space-y-4 mb-14 flex-1">
                            <li class="flex items-center gap-3 text-sm text-slate-600 dark:text-white/70 transition-colors duration-500">
                                <span class="w-1.5 h-1.5 rounded-full bg-[#3B82F6]"></span> Context-before-task structure
                            </li>
                            <li class="flex items-center gap-3 text-sm text-slate-600 dark:text-white/70 transition-colors duration-500">
                                <span class="w-1.5 h-1.5 rounded-full bg-[#3B82F6]"></span> Grounding cues emphasis
                            </li>
                            <li class="flex items-center gap-3 text-sm text-slate-600 dark:text-white/70 transition-colors duration-500">
                                <span class="w-1.5 h-1.5 rounded-full bg-[#3B82F6]"></span> Multimodal references
                            </li>
                            <li class="flex items-center gap-3 text-sm text-slate-800 dark:text-white font-medium transition-colors duration-500">
                                <span class="w-1.5 h-1.5 rounded-full bg-[#3B82F6]"></span> 25% weight on model alignment
                            </li>
                        </ul>
                        
                        <div class="h-1 w-full bg-slate-100 dark:bg-white/5 rounded-full flex overflow-hidden transition-colors duration-500">
                            <div class="h-full bg-primary w-[15%]"></div>
                            <div class="h-full bg-[#3B82F6] w-[15%]"></div>
                            <div class="h-full bg-amber-400 w-[30%]"></div>
                            <div class="h-full bg-slate-200 dark:bg-white/10 w-[40%]"></div>
                        </div>
                    </div>

                </div>
            </div>
        </section>"""

start_token = "<!-- ── Model Showcase ───────────────────────────────────── -->"
end_token = "<!-- ── How It Works ─────────────────────────────────────── -->"

start_idx = html.find(start_token)
end_idx = html.find(end_token)

if start_idx != -1 and end_idx != -1:
    new_html = html[:start_idx] + new_section + "\n\n        " + html[end_idx:]
    with open(html_path, 'w', encoding='utf-8') as f:
        f.write(new_html)
    print("Replaced Model Showcase successfully")
else:
    print("Tokens not found.")

# Update CSS path
css_path = 'f:/PROMPTLAB/public/css/landing.css'
with open(css_path, 'r', encoding='utf-8') as f:
    css = f.read()

if '.grid-overlay' not in css:
    css_append = """

.grid-overlay {
    background-size: 40px 40px;
    background-image: 
        linear-gradient(to right, rgba(0, 0, 0, 0.05) 1px, transparent 1px),
        linear-gradient(to bottom, rgba(0, 0, 0, 0.05) 1px, transparent 1px);
}

.dark .grid-overlay {
    background-image: 
        linear-gradient(to right, rgba(255, 255, 255, 0.03) 1px, transparent 1px),
        linear-gradient(to bottom, rgba(255, 255, 255, 0.03) 1px, transparent 1px);
}
"""
    with open(css_path, 'a', encoding='utf-8') as f:
        f.write(css_append)
    print("CSS grid-overlay added successfully")
else:
    print("CSS already has grid-overlay")

File: update_landing.js
const fs = require('fs');
const path = require('path');

const htmlPath = path.join('f:', 'PROMPTLAB', 'public', 'landing.html');
let html = fs.readFileSync(htmlPath, 'utf8');

const newSection = `        <!-- ── How It Works ─────────────────────────────────────── -->
        <section class="relative z-10 py-24 px-4 bg-white dark:bg-[#000000] transition-colors duration-500 min-h-[800px]"
            id="how-it-works">
            <div class="absolute inset-0 grid-overlay-works pointer-events-none opacity-10 dark:opacity-40 transition-opacity duration-500"></div>
            <div class="absolute top-1/4 left-1/2 -translate-x-1/2 w-[800px] h-[500px] bg-primary/5 dark:bg-primary/5 blur-[120px] rounded-full pointer-events-none transition-colors duration-500"></div>
            
            <div class="relative z-10 max-w-7xl mx-auto flex flex-col items-center">
                <div class="flex flex-col gap-4 text-center items-center mb-16">
                    <div
                        class="inline-flex items-center gap-2 px-3 py-1 rounded-full bg-slate-100 dark:bg-white/5 border border-slate-200 dark:border-white/10 text-xs font-bold tracking-widest text-slate-500 dark:text-white/60 uppercase transition-colors duration-500">
                        Process
                    </div>
                    <h2
                        class="text-4xl md:text-5xl lg:text-6xl font-black tracking-tighter text-slate-900 dark:text-white transition-colors duration-500 mb-8 w-full block">
                        How <span
                            class="text-primary dark:breathing-glow transition-all duration-500">PromptLab</span>
                        Works
                    </h2>
                </div>

                <div class="grid grid-cols-1 md:grid-cols-3 gap-12 md:gap-8 w-full relative">
                    <!-- Connector Line (Desktop only) -->
                    <div
                        class="hidden md:block absolute top-[32px] left-[15%] right-[15%] h-px connector-line transition-all duration-500">
                    </div>

                    <!-- Step 1 -->
                    <div class="flex flex-col items-center text-center relative z-10 group">
                        <div
                            class="size-16 rounded-lg bg-white dark:bg-black border border-slate-200 dark:border-primary/30 shadow-lg dark:shadow-[0_0_20px_rgba(255,77,77,0.15)] flex items-center justify-center mb-6 text-xl font-bold font-mono text-primary group-hover:scale-110 transition-transform duration-300 dark:card-glass-works">
                            01
                        </div>
                        <h3
                            class="text-xl md:text-2xl font-bold text-slate-900 dark:text-white mb-3 transition-colors duration-500">
                            Paste Your Prompt</h3>
                        <p class="text-slate-500 dark:text-white/40 text-sm max-w-xs transition-colors duration-500 leading-relaxed">
                            Enter any prompt you'd send to ChatGPT, Claude, or Gemini. Choose your target model.
                        </p>
                    </div>

                    <!-- Step 2 -->
                    <div class="flex flex-col items-center text-center relative z-10 group">
                        <div
                            class="size-16 rounded-lg bg-white dark:bg-black border border-slate-200 dark:border-primary/30 shadow-lg dark:shadow-[0_0_20px_rgba(255,77,77,0.15)] flex items-center justify-center mb-6 text-xl font-bold font-mono text-primary group-hover:scale-110 transition-transform duration-300 dark:card-glass-works">
                            02
                        </div>
                        <h3
                            class="text-xl md:text-2xl font-bold text-slate-900 dark:text-white mb-3 transition-colors duration-500">
                            Blueprint Comparison</h3>
                        <p class="text-slate-500 dark:text-white/40 text-sm max-w-xs transition-colors duration-500 leading-relaxed">
                            PromptLab compares your prompt against the ideal structural blueprint for that model × task.
                        </p>
                    </div>

                    <!-- Step 3 -->
                    <div class="flex flex-col items-center text-center relative z-10 group">
                        <div
                            class="size-16 rounded-lg bg-white dark:bg-black border border-slate-200 dark:border-primary/30 shadow-lg dark:shadow-[0_0_20px_rgba(255,77,77,0.15)] flex items-center justify-center mb-6 text-xl font-bold font-mono text-primary group-hover:scale-110 transition-transform duration-300 dark:card-glass-works">
                            03
                        </div>
                        <h3
                            class="text-xl md:text-2xl font-bold text-slate-900 dark:text-white mb-3 transition-colors duration-500">
                            Actionable Feedback</h3>
                        <p class="text-slate-500 dark:text-white/40 text-sm max-w-xs transition-colors duration-500 leading-relaxed">
                            See scores, issues, suggestions, and a model-specific optimization checklist — then improve.
                        </p>
                    </div>
                </div>
            </div>
        </section>`;

const startToken = "<!-- ── How It Works ─────────────────────────────────────── -->";
const endToken = "<!-- ── Testimonials Section ─────────────────────────────────── -->";

const startIdx = html.indexOf(startToken);
const endIdx = html.indexOf(endToken);

if (startIdx !== -1 && endIdx !== -1) {
    const newHtml = html.substring(0, startIdx) + newSection + "\\n\\n        " + html.substring(endIdx);
    fs.writeFileSync(htmlPath, newHtml, 'utf8');
    console.log("Replaced How It Works section successfully");
} else {
    console.log("Tokens not found.");
}

// Update CSS path
const cssPath = path.join('f:', 'PROMPTLAB', 'public', 'css', 'landing.css');
let css = fs.readFileSync(cssPath, 'utf8');

if (!css.includes('.grid-overlay-works')) {
    const cssAppend = `\n
.grid-overlay-works {
    background-size: 60px 60px;
    background-image: 
        linear-gradient(to right, rgba(0, 0, 0, 0.05) 1px, transparent 1px);
}

.dark .grid-overlay-works {
    background-image: 
        linear-gradient(to right, rgba(255, 77, 77, 0.05) 1px, transparent 1px);
}

.dark .card-glass-works {
    background: rgba(10, 10, 10, 0.6);
    backdrop-filter: blur(12px);
    border: 1px solid rgba(255, 77, 77, 0.1);
}

.connector-line {
    background: linear-gradient(90deg, transparent, rgba(0, 0, 0, 0.1), transparent);
}

.dark .connector-line {
    background: linear-gradient(90deg, transparent, rgba(255, 77, 77, 0.3), transparent);
    box-shadow: 0 0 15px rgba(255, 77, 77, 0.1);
}

.dark .breathing-glow {
    animation: breathe 3s ease-in-out infinite;
}

@keyframes breathe {
    0%, 100% { text-shadow: 0 0 10px rgba(255, 77, 77, 0.4); }
    50% { text-shadow: 0 0 25px rgba(255, 77, 77, 0.8), 0 0 45px rgba(255, 77, 77, 0.4); }
}
`;
    fs.appendFileSync(cssPath, cssAppend, 'utf8');
    console.log("CSS How-It-Works utilities added successfully");
} else {
    console.log("CSS already has How-It-Works utilities");
}
